### DISCUSSION TITLE
[Laravel] Database transactions in controller not working properly in Pest test

### DISCUSSION THREAD

**Discussion by Jamesking56** [pestphp/pest#493] [Q&A]
When using Laravel's `DB::transaction()` method, Pest incorrectly perceives that no records are created.

For instance, using this `Pest.php` file:

```php
uses(TestCase::class, LazilyRefreshDatabase::class)->in('Feature');
uses(TestCase::class)->in('Unit');
```

And creating a Controller method that does multiple queries inside of `DB::transaction()`:

```php
DB::transaction(function() {
    // Queries in here
});
```

Then hitting the controller in a Pest test and checking that the database has those records created:

```php
it('tests transactions', function() {
    postJson('/api/some-route', $data, ['Authorization' => 'Bearer ' . $accessToken])
        ->assertStatus(\Illuminate\Http\Response::HTTP_CREATED)
        ->assertJson([
            'success' => true
        ]);

    assertDatabaseCount('some-table', 3);
});
```

Seems to fail, stating that no records exist in the table. Really not sure why, the code works fine and there is no exception logged 🤷‍♂️

**Comment by pildit** [comment]
There seems to be a problem with DB transaction, even when using phpUnit. 
I wasn't able to wrap all the tests from a class in a transaction, tried (almost) everything ... The DB::commit simply didn't work..

**Comment by briantuju** [comment]
Any help on this?

**Comment by viicslen** [comment]
Having the same issue as well when using `uses(DatabaseTransactions::class);` 

```php
use App\Models\Customers\Customer;
use Illuminate\Foundation\Testing\DatabaseTransactions;

use function Pest\Laravel\postJson;

uses(DatabaseTransactions::class);

it('should test if customer is deactivated successfully', function () {
    $customer = Customer::factory()->create();
    
    postJson(route('api.v2.contacts.deactivate'), ['contact_id' => $customer->id])
        ->assertSuccessful()
        ->assertJsonPath('message', 'Contact deactivated successfully');
    
    expect($customer->fresh()->deleted_at)->not()->toBeNull();
});
```

**Comment by Luke-Henry-RedCat** [comment]
I'm having the same issue. Any idea when or if this might be addressed? Are there any known workarounds?
